nefertari/elasticsearch.py

```
@@ -758,38 +758,45 @@ def build_search_params(self, params):
-                    'query': {
-                        'query_string': {
-                            'query': query_string
-                        }
-                    }
-                }
+                query['must'].append({'query_string': {'query': query_string}})
+
+            if analyzed_terms:
+                query['must'].append(analyzed_terms)
+
+            if query['must']:
+                _params['body'] = {'query': {'bool': query}}
             else:
-                _params['body'] = {"query": {"match_all": {}}}
-        else:
-            _params['body'] = params['body']
+                _params['body'] = {'query': {'match_all': {}}}
+
+        if 'body' in params:
+            raise JHTTPUnprocessableEntity('Illegal parameter "body"')
+
         if '_limit' not in params:
             params['_limit'] = self.api.count(index=self.index_name)['count']
-
         _params['from_'], _params['size'] = process_limit(
             params.get('_start', None),
             params.get('_page', None),
             params['_limit'])
 
         if 'es_q' in params:
-            if 'query' in _params['body']:
-                try:
-                    _params['body']['query'] = compile_es_query(params)
-                except Exception:
-                    raise JHTTPBadRequest('Bad query string for {params}'
-                                          .format(
-                                                  params=_params['body']['query']['query_string']['query']))
-
-                log.debug('Parsed ES request body {body}'.format(body=_params['body']['query']))
+            _params['body'] = {}
+
+            try:
+                _params['body']['query'] = compile_es_query(params)
+            except Exception:
+                raise JHTTPBadRequest('Bad query string for {params}'
+                                        .format(
+                                                params=_params['body']['query']['query_string']['query']))
```

Why not just let the exception bubble up by itself?

---

I think there didn't informant for user to know that ValueError raised